src/rp2/plugin/report/abstract_ods_generator.py

```diff
@@ -54,7 +55,14 @@ def _initialize_output_file(
         if Path(output_file_path).exists():
             output_file_path.unlink()
 
-        template_path: str = str(Path(os.path.dirname(__file__)).absolute() / Path("".join(["data/template_", country.country_iso_code, ".ods"])))
+        if len(template_file_prefix) > 0 and template_file_prefix.startswith("_") is False:
+            template_file_prefix = "_" + template_file_prefix
+        template_path: str = str(
+            Path(os.path.dirname(__file__)).absolute() / Path("".join(["data/template", template_file_prefix, "_", country.country_iso_code, ".ods"]))
+        )
+        if Path(template_path).exists() is False:
+            raise Exception(f"Error: template file '{template_path}' not found.")
+
```

Comment:
The main reason I put in the code this way was because I wasn't sure if you wanted to leave the country code bit in the generator, or push that responsibility up to the individual reports to do or enforce. I opted for the lighter touch, but if that works for you I can do the change and put the country code bit in the other reports.

Alternatively, if you're inclined, you can update the generator and  2 reports how you'd like it to work, and I can follow suit.

Reply:
Gotcha, good point: let me work on this. I'll update the code shortly.

Reply:
I uploaded the new logic: now plugins pass the full path to the template file to `_initialize_output_file()`

Reply:
Excellent, got it and applied.